compiler/rustc_ty_utils/src/ty.rs

```diff
@@ -373,8 +373,8 @@ fn well_formed_types_in_env<'tcx>(
     let input_clauses = inputs.into_iter().filter_map(|arg| {
         match arg.unpack() {
             GenericArgKind::Type(ty) => {
-                let binder = Binder::dummy(PredicateAtom::TypeWellFormedFromEnv(ty));
-                Some(tcx.mk_predicate(PredicateKind::ForAll(binder)))
+                let binder = Binder::dummy(PredicateKind::TypeWellFormedFromEnv(ty));
+                Some(tcx.mk_predicate(binder))
```

Comment:
```suggestion
                Some(tcx.mk_predicate(PredicateKind::TypeWellFormedFromEnv(ty)))
```
nit

Reply:
I'm confused here. `mk_predicate` takes a `Binder<PredicateKind<'tcx>>`. Do you mean `PredicateKind::TypeWellFormedFromEnv(ty).to_predicate()`?

Reply:
yeah, it's `to_predicate`, sry :sweat_smile:

Reply:
In that case, that's actually a pattern I'd like to remove at some point (in favor of the `Binder<PredicateKind>` impl; the point being to try to remove implicit `Binder::dummy`s). Since these are semantically the same and this is only a nit, I'm going to leave as-is if that's okay with you.